### PR TITLE
Fix query error from hive external table when boolean data type as partition key

### LINKED ISSUE #4434: query error from hive external table when hive table with boolean data type as partition key
### Steps to reproduce the behavior (Required)
1. create hive table and insert data:
```
 create table hive_hdfs_par_date_boolean(
 col_tinyint tinyint comment "column tinyint"
 ,col_int int comment "column int"
 ,col_integer integer comment "column integer"
 ,col_bigint bigint comment "column bigint"
 ,col_float float comment "column float"
 ,col_double double comment "column double"
 ,col_double_precision double precision comment "column double precision"
 ,col_decimal decimal comment "column decimal"
 ,col_timestamp timestamp comment "column timestamp"
 ,col_string string comment "column string"
 ,col_varchar varchar(100) comment "column varchar"
 ,col_binary binary comment "column binary"
 ,col_char char(100) comment "column char"
 ,col_array array<int> comment "column array"
 )
 partitioned by(col_date date,col_boolean boolean)
 clustered by(col_int) sorted by(col_int) into 32 buckets
 row format delimited fields terminated by ','
 stored as orc
 tblproperties("orc.compress"="none");
 insert into hive_hdfs_par_date_boolean partition(col_date = '2021-01-01',col_boolean=1) select 1,1,1,1,1.001,10.1,110.1,1110.1,'2021-10-30 12:10:23','hello world','hi hello','1110001010101011001001','nihao',array(1,10,100);
 insert into hive_hdfs_par_date_boolean partition(col_date = '2021-01-01',col_boolean=0) select 1,1,1,1,1.001,10.1,110.1,1110.1,'2021-10-30 12:10:23','hello world','hi hello','1110001010101011001001','nihao',array(1,10,100);
 insert into hive_hdfs_par_date_boolean partition(col_date = '2021-02-01',col_boolean=1) select 2,2,2,2,2.001,20.1,210.1,2110.2,'2021-02-01 12:10:23','hello world','hi hello','1110001010101011001001','nihao',array(2,20,200);
 insert into hive_hdfs_par_date_boolean partition(col_date = '2021-02-01',col_boolean=0) select 2,2,2,2,2.001,20.1,210.1,2110.2,'2021-02-01 12:10:23','hello world','hi hello','1110001010101011001001','nihao',array(2,20,200);
```
data path in hdfs:
```
[dorisdb@daily01 ~]$ hdfs dfs -ls /user/hive/warehouse/tn_test.db/hive_hdfs_par_date_boolean/col_date=2021-01-01
Found 2 items
drwxrwxrwx   - dorisdb supergroup          0 2022-03-25 14:09 /user/hive/warehouse/tn_test.db/hive_hdfs_par_date_boolean/col_date=2021-01-01/col_boolean=false
drwxrwxrwx   - dorisdb supergroup          0 2022-03-25 14:09 /user/hive/warehouse/tn_test.db/hive_hdfs_par_date_boolean/col_date=2021-01-01/col_boolean=true
```

2. create hive external table:
```
 create external table ex_hive_par_date_boolean ( 
 col_tinyint tinyint null comment "column tinyint"
 ,col_int int comment "column int"
 ,col_integer int comment "column integer"
 ,col_bigint bigint comment "column bigint"
 ,col_float float comment "column float"
 ,col_double double comment "column double"
 ,col_double_precision double comment "column double precision"
 ,col_decimal decimal comment "column decimal"
 ,col_timestamp datetime comment "column timestamp"
 ,col_string varchar(200) comment "column string"
 ,col_varchar varchar(2) comment "column varchar"
 ,col_binary varchar(200) comment "column binary"
 ,col_char varchar(100) comment "column char"
 ,col_boolean boolean null comment "column boolean"
 ,col_date date null )
 ENGINE=hive 
 properties ( 
 "resource" = "hive2", 
 "table" = "hive_hdfs_par_date_boolean", 
 "database" = "tn_test");
```

3. select from hive external table:
```
mysql> select * from ex_hive_par_date_boolean;
ERROR 1064 (HY000): com.starrocks.common.DdlException: get partition detail failed: com.starrocks.common.DdlException: get hive partition meta data failed: partition not exists, partValues: 2021-01-01,0
```

### Expected behavior (Required)
return the right result

### Real behavior (Required)
return error

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+------------------------+
| current_version()      |
+------------------------+
| QA_TEST_MASTER a1ae0dc |
+------------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by adzfolc**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #4434 

## Problem Summary(Required) ：

This PR implement bug fix of query from hive external table when hive table with boolean data type as partition key.
The bug is fixed by optimizing the logic of getting partition values.

### CODE REVIEW

**Comment by DorianZheng** [comment]
@adzfolc should add UT for this

**Comment by DorianZheng** [comment]
@adzfolc thanks for the contribution, well done.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|

**Review by imay — DISMISSED** [review]
LGTM

**Review by DorianZheng — DISMISSED** [review]
LGTM

**Review by caneGuy — DISMISSED** [review]

**Comment by caneGuy** [comment]
run starrocks_clang-format

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|

File: fe/fe-core/src/test/java/com/starrocks/external/hive/UtilsTest.java [hunk 1]
```diff
@@ -35,6 +36,12 @@ public void testGetPartitionValues() throws Exception {
         List<String> values = Lists.newArrayList("1", "a", "3.0", HiveMetaClient.PARTITION_NULL_VALUE);
         PartitionKey partitionKey = Utils.createPartitionKey(values, partColumns);
         Assert.assertEquals(values, Utils.getPartitionValues(partitionKey));
+
+        List<Column> partColumns1 = Lists.newArrayList(new Column("k1", Type.DATE), new Column("k2", Type.BOOLEAN));
+        PartitionKey partitionKey1 = Utils.createPartitionKey(Lists.newArrayList("2021-01-01", "false"), partColumns1);
+        List<String> partValues1 = Utils.getPartitionValues(partitionKey1);
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/external/hive/UtilsTest.java:42]: add isHudiTable condition.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 5 / 5 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|5|5|100.00%|

File: fe/fe-core/src/test/java/com/starrocks/external/hive/UtilsTest.java [hunk 2]
```diff
@@ -34,7 +35,13 @@ public void testCreatePartitionKey() throws Exception {
     public void testGetPartitionValues() throws Exception {
         List<String> values = Lists.newArrayList("1", "a", "3.0", HiveMetaClient.PARTITION_NULL_VALUE);
         PartitionKey partitionKey = Utils.createPartitionKey(values, partColumns);
-        Assert.assertEquals(values, Utils.getPartitionValues(partitionKey));
+        Assert.assertEquals(values, Utils.getPartitionValues(partitionKey, false));
+
+        List<Column> partColumns1 = Lists.newArrayList(new Column("k1", Type.DATE), new Column("k2", Type.BOOLEAN));
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/external/hive/UtilsTest.java:40]: LGTM

**Review by stephen-shelby — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/external/hive/Utils.java [hunk 3]
```diff
@@ -62,6 +63,11 @@ public static PartitionKey createPartitionKey(List<String> values, List<Column>
                 } else {
                     values.add(HiveMetaClient.PARTITION_NULL_VALUE);
                 }
+            } else if (value instanceof BoolLiteral) {
+                if (!isHudiTable) {
```
Comment (caneGuy) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/Utils.java:67]: why check isHudiTable here?

**Review by caneGuy — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 3 / 3 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/external/hive/Utils.java|3|3|100.00%|

**Pull request merged**